File: swiggy-frontend/src/services/orderService.js
import api from "./api";

export const placeOrder = (orderData) => {
    return api.post("/orders/place", orderData);
};

export const getMyOrders = () => {
    return api.get("/orders/my");
};

export const getOrderById = (orderId) => {
    return api.get(`/orders/${orderId}`);
};

export const cancelOrder = (orderId) => {
    return api.delete(`/orders/${orderId}`);
};

export const trackOrder = (orderId) => {
    return api.get(`/orders/${orderId}/tracking`);
};

File: swiggy-backend/src/main/java/com/swiggy/swiggy_backend/service/OrderService.java
package com.swiggy.swiggy_backend.service;

import java.util.List;

import com.swiggy.swiggy_backend.dto.OrderResponse;
import com.swiggy.swiggy_backend.dto.OrderStatusRequest;
import com.swiggy.swiggy_backend.dto.OrderTrackingResponse;
import com.swiggy.swiggy_backend.dto.PlaceOrderRequest;
import com.swiggy.swiggy_backend.entity.OrderStatus;

public interface OrderService {

    OrderResponse placeOrder(PlaceOrderRequest request);

    OrderResponse getOrderById(Long orderId);

    List<OrderResponse> getMyOrders();

    OrderResponse updateOrderStatus(Long orderId, OrderStatus status);

    void cancelOrder(Long orderId);
    void updateOrderStatus(Long orderId, OrderStatusRequest request);

    OrderTrackingResponse trackOrder(Long orderId);

}